### ISSUE TITLE
Python SSL support is unavailable

### ISSUE THREAD

**Issue by imrn** [void-linux/void-packages#21994]
I don't know if it is libressl related but python ssl support is not working.
See: #20935

```python3
import http.client
conn = http.client.HTTPSConnection("www.python.org")
conn.request("GET", "/")
r1 = conn.getresponse()
print(r1.status, r1.reason)
```
terminates with:
AttributeError: module 'http.client' has no attribute 'HTTPSConnection'

Similarly:
(Install python3-requests first)
```python3
import requests
r = requests.get('https://www.python.org')
print(r.text)
```
requests.exceptions.SSLError: HTTPSConnectionPool(host='https://www.python.org', port=443): Max retries exceeded with url: / (Caused by SSLError("Can't connect to HTTPS URL because the SSL module is not available."))

**Comment by Hoshpak** [comment]
Works fine for me:

```
>>> import http.client
>>> conn = http.client.HTTPSConnection("www.python.org")
>>> conn.request("GET", "/")
>>> r1 = conn.getresponse()
>>> print(r1.status, r1.reason)
200 OK
```

Please supply some more pieces of information about your system as laid out in the issue template.

**Comment by q66** [comment]
also works for me

**Comment by sgn** [comment]
I removed `python3` and reinstalled it.
Your code works for me.

**Comment by fosslinux** [comment]
Also works for me. What does `xbps-query -l` look like? Are you fully up-to-date?

**Comment by imrn** [comment]
Thanks for the input. I'm trying to do a full update but unfortunely currently alpha.de.repo.voidlinux.org throughput is around 20-25kb/s at my location. No problems for other routes.) Anyway I'll post more results after the update.

**Comment by imrn** [comment]
I'm using linux5.6. Other than that, the installation is a regular and up-to-date one without any customization and the problem persists.

xuname: Void 5.6.13_1 x86_64 GenuineIntel uptodate rrFFF
ii python3-3.8.2_1   Python programming language (3.8 series)
(Reinstalled python3. Nothing has changed.)

Should I post "xbps-query -l" here? It is long.

**Comment by Chocimier** [comment]
What's output of `python-m ssl` ?

**Comment by imrn** [comment]
Output of "python3 -m ssl" is nothing.
Just an empty new line.

**Comment by Chocimier** [comment]
Should work then. Are you using virtualenv, non-system interpreters... ?

**Comment by imrn** [comment]
No. I just use void provided packages. No customization as I said.

This also works without any problem.
"openssl s_client -connect www.python.org:443"

**Comment by imrn** [comment]
This is a regular text file and runs on terminal. Nothing fancy.

```python
#!/bin/python3
import http.client
conn = http.client.HTTPSConnection("www.python.org")
conn.request("GET", "/")
r1 = conn.getresponse()
print(r1.status, r1.reason)
```

**Comment by imrn** [comment]
Same file is run by different users of the machine (with unrelated environments, homedirs, confs, etc) all having the same result.

**Comment by imrn** [comment]
I've even created a brand new machineuser to eliminate any user specific conf,  env, etc interference and run the script there having the same result.

**Comment by imrn** [comment]
Non-SSL version just works. (Note 'HTTP' below. 'S'. is absent)
I guess the problem is local to something between python and libressl.
But I can not understand why it is specific to my machine.
How can I find it? Any ideas?

```python3
#!/bin/python3
import http.client
conn = http.client.HTTPConnection("www.python.org")
conn.request("GET", "/")
r1 = conn.getresponse()
print(r1.status, r1.reason)```

**Comment by imrn** [comment]
Found it!. I discovered that I had an experimental obscure "ssl.py" within same directory which also imports many classes from python's own "ssl".

Interestingly as above examples has no direct "import ssl" statement, somehow it interferes with them and their tracebacks provide no clues about it! Traceback does not even mention that obscure file and its classes!!

Luckly, as I decided to experiment with some more low level stuff below and analyze the trace back, I've discovered it. This time that obscure file surfaced in the traceback.

Bad luck, mixed with python's "marvelous" import machinery and exception handing creates interesting and hard to find bugs.

Thank you all for your comments for eliminating above possibilities so that I could focus on proper problem region. Just closing.

```python3
import socket
import ssl

hostname = 'www.python.org'
context = ssl.create_default_context()
with socket.create_connection((hostname, 443)) as sock:
    with context.wrap_socket(sock, server_hostname=hostname) as ssock:
        print(ssock.version())
```
